### DISCUSSION TITLE
undefined reference to `_Ux86_64_getcontext' - Cannot compile simple libunwind program on Debian bullseye and bookworm

### DISCUSSION THREAD

**Discussion by sopho22** [libunwind/libunwind#683] [Q&A] [answered]
Hi, 
I'm trying to compile a simple test program:
---
#include <stdio.h>
#include <stdlib.h>
#include <libunwind.h>

int main(int argc, char **argv)
{
	unw_cursor_t cursor;
	unw_context_t context;
	unw_getcontext(&context);
	unw_init_local(&cursor, &context);
	return 1;
}
---
using gcc command line:
gcc -lunwind main.c -o test

under Debian, but get following compile errors:
/usr/bin/ld: /tmp/cczoMgbs.o: in function `main':
main.c:(.text+0x23): undefined reference to `_Ux86_64_getcontext'
/usr/bin/ld: main.c:(.text+0x3c): undefined reference to `_Ux86_64_init_local'
collect2: error: ld returned 1 exit status

I've installed the libunwind-dev package and tried it on two different machines (one clean install) but cannot compile.
Am I doing something wrong or is libunwind broken in Debian?

Thanks,
Soph

**Comment by bertwesarg** [comment, marked as answer]
Two ideas:

1. Put `-l` flags last

```console
$ gcc main.c -o test -lunwind
```

2. Use `UNW_LOCAL_ONLY`

```diff
 #include <stdlib.h>
+#define UNW_LOCAL_ONLY
 #include <libunwind.h>
```

**Reply by sopho22** [reply]
#define UNW_LOCAL_ONLY
made it work.Thanks a lot!
